### PR TITLE
Honeywell cleanup

### PR DESCRIPTION
**Pull request by jhenkens**
## Proposed change

Corresponding documentation change to https://github.com/home-assistant/core/pull/36456/

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [x] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/36456
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by jhenkens** [comment]
I've messed up this branch - waiting for local pull to rebase. Darn web IDEs

**Review by frenck — DISMISSED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

**Comment by MartinHjelmare** [comment]
Please add thermostat and location to the configuration documentation.

**Comment by jhenkens** [comment]
I'm so sorry - left it in staging on accident when I rerequested re-review on the other one.

**Comment by MartinHjelmare** [comment]
Maybe those should be added in another PR that targets `current` branch, since they are already existing in the current integration?

**Comment by jhenkens** [comment]
My understanding could be wrong, but without the changes in dev that actually expose them during the config schema setup, these values are not assignable. Thus, documentation in dev makes sense.

**Comment by MartinHjelmare** [comment]
Right, you're correct. I was thinking that the schema did allow them without validating them, but I've checked the schema on master and it's not allowing them.

**Comment by jhenkens** [comment]
Anything blocking this MR at this point? I'd like to free my mental slots so I can being progress on the next MRs.

**Comment by MartinHjelmare** [comment]
This PR still has https://github.com/home-assistant/core/pull/36329 as parent. Should we close that PR and change parent here to https://github.com/home-assistant/core/pull/36456?

**Comment by jhenkens** [comment]
Ah my mistake. The parent was changed. I’ll update.

**Comment by jhenkens** [comment]
Updated the parent, and closed the other PR. I'll continue my changes in subsequent PRs, piecemeal.

File: source/_integrations/honeywell.markdown [hunk 1]
```diff
@@ -26,23 +26,17 @@ climate:
   - platform: honeywell
     username: YOUR_USERNAME
     password: YOUR_PASSWORD
-    region: us
 ```
 
 {% configuration %}
 username:
-  description: Email address of an account with access the TCC website for your region.
+  description: Email address of an account with access the TCC website
```
Comment (MartinHjelmare) [inline source/_integrations/honeywell.markdown]:
```suggestion
  description: Email address of an account with access to the TCC website.
```

File: source/_integrations/honeywell.markdown [hunk 2]
```diff
@@ -53,4 +47,12 @@ away_heat_temperature:
   required: false
   default: 61
   type: integer
+thermostat:
+  description: ID of thermostat to restrict intgration to
```
Comment (MartinHjelmare) [inline source/_integrations/honeywell.markdown]:
```suggestion
  description: ID of thermostat to restrict the integration to.
```

File: source/_integrations/honeywell.markdown [hunk 3]
```diff
@@ -53,4 +47,12 @@ away_heat_temperature:
...
+  required: false
+  type: string
+location:
+  description: ID of location to restrict intgration to
```
Comment (MartinHjelmare) [inline source/_integrations/honeywell.markdown]:
```suggestion
  description: ID of location to restrict the integration to.
```

**Review by frenck — APPROVED** [review]
Thanks, @jhenkens 👍

**Pull request merged**